platform/darwin/src/MGLBackgroundStyleLayer.mm

```diff
@@ -35,6 +35,10 @@ - (instancetype)initWithIdentifier:(NSString *)identifier
 
 - (void)addToMapView:(MGLMapView *)mapView
 {
+    if (_pendingLayer == nullptr) {
+        [NSException raise:NSInternalInconsistencyException format:@"Adding the same instance to the style more than once is invalid."];
```

Comment:
If the developer is accidentally readding an instance to a style, they’ll want to know which one. We can describe the layer and style or map view here.